hyperspy/signal.py

```diff
@@ -4941,6 +5061,236 @@ def _plot_permanent_markers(self):
+
+        title : string
+            Title of the plot.
+
+        with_centers : bool
+            If True, also returns figure(s) with the cluster centers for the
+            given cluster_ids.
+
+        cmap : matplotlib colormap
+            The colormap used for the factor image, or for peak
+            characteristics, the colormap used for the scatter plot of
+            some peak characteristic.
+
+        no_nans : bool
+            If True, removes NaN's from the loading plots.
+
+        per_row : int
+            the number of plots in each row, when the same_window
+            parameter is True.
+
+        axes_decor : {'all', 'ticks', 'off', None}, optional
+            Controls how the axes are displayed on each image; default is 'all'
+            If 'all', both ticks and axis labels will be shown
+            If 'ticks', no axis labels will be shown, but ticks/labels will
+            If 'off', all decorations and frame will be disabled
+            If None, no axis decorations will be shown, but ticks/frame will
+
+        See Also
+        --------
+        plot_cluster_centers, plot_cluster_results.
+
+        """
+        if self.axes_manager.navigation_dimension > 2:
+            raise NotImplementedError("This method cannot plot labels of "
+                                      "dimension higher than 2."
+                                      "You can use "
+                                      "`plot_cluster_results` instead.")
+        if same_window is None:
+            same_window = True
+        labels = self.learning_results.cluster_labels
+        if with_centers:
+            centers = self.learning_results.cluster_centers.T
+        else:
+            centers = None
+
+        if cluster_ids is None:
+            if self.learning_results.number_of_clusters:
+                cluster_ids = self.learning_results.number_of_clusters
+            else:
+                raise ValueError(
+                    "Please provide the number of cluster labels"
+                    "to plot via the ``cluster_ids`` argument")
+
+        title = _change_API_comp_label(title, comp_label)
+        if title is None:
+            title = self._get_plot_title('Cluster labels of',
+                                         same_window)
+
+        return self._plot_loadings(
+            labels,
+            comp_ids=cluster_ids,
+            with_factors=with_centers,
+            factors=centers,
+            same_window=same_window,
+            comp_label=title,
+            cmap=cmap,
+            no_nans=no_nans,
+            per_row=per_row,
+            axes_decor=axes_decor)        
+
+        
+    def plot_cluster_results(self,
+                             centers_navigator="auto",
+                             labels_navigator="auto",
+                             centers_dim=2,
+                             labels_dim=2
+                             ):
+        """Plot the cluster labels and centers.
```

Comment:
Similar as above, add option to plot a composite label maps? In case of `Signal1D` signal, it should be possible to plot all spectra on the plot and with line color matching the composite maps.

Reply:
I was trying to minimise the code added and avoid more plot functions. The other issue is normalisation - clustered data from a background and sample area can have very different intensities. To plot them together they need to have the same scaling and this is difficult to do correctly  - e.g. for EELS or XANES